cmd/osbuild-image-tests/main_test.go

```diff
@@ -273,6 +273,42 @@ func testBootUsingAWS(t *testing.T, imagePath string) {
 	require.NoError(t, err)
 }
 
+func testBootUsingAzure(t *testing.T, imagePath string) {
+	creds, err := getAzureCredentialsFromEnv()
+	require.NoError(t, err)
+
+	// if no credentials are given, fall back to qemu
+	if creds == nil {
+		log.Print("no Azure credentials given, falling back to booting using qemu")
+		testBootUsingQemu(t, imagePath)
+		return
+	}
+
+	imageId, err := generateRandomString("")
```

Comment:
Same comment as previously for your TODO log. Why not use the image UUID instead of a random string? Will make it easier to find on the remote side too.

Reply:
I agree I'm planning to change it, this is just a result of copy-paste.

Reply:
Done